### DISCUSSION TITLE
Nuxt 3 and Apollo

### DISCUSSION THREAD

**Discussion by winnysan** [nuxt/nuxt#16666] [Questions] [answered]
Hi guys, is information whether or not `@nuxtjs/apollo` module will be made for Nuxt 3?

I created my own plugin with `@apollo/client` and `@vue/apollo-composable` .
According to the documentation `https://v4.apollo.vuejs.org/guide-composable/`, queries, mutations and cache work very well for me.
But I can't make it work `subscription`. I using Laravel Lighthouse as backend.

Is there any idea how to do it right? Only for information... I'm not a programmer, I'm a home hobbyist.

Here is my plugin code:

```
import { defineNuxtPlugin } from '#app'
import { ApolloClient, createHttpLink, InMemoryCache } from '@apollo/client/core'
import { provideApolloClient } from '@vue/apollo-composable'

import { HttpLink, split } from "@apollo/client/core"
import { WebSocketLink } from "@apollo/client/link/ws"
import { getMainDefinition } from "@apollo/client/utilities"

export default defineNuxtPlugin((nuxtApp) => {
  const token = useCookie('access-token').value

  const getHeaders = () => {
    const headers = {}
    if (token) {
      headers.authorization = `Bearer ${token}`
    }
    return headers
  }

  const httpLink = createHttpLink({ // HttpLink not working
    uri: 'https://api.example.com/graphql',
    headers: getHeaders()
  })

  const wsLink = process.browser ? new WebSocketLink({
    uri: 'ws://api.example.com:6001/',
    options: {
      reconnect: true
    }
  }) : null

  const link = process.browser ? split( // not working on SSR
    ({ query }) => {
      const definition = getMainDefinition(query)
      return (
        definition.kind === "OperationDefinition" &&
        definition.operation === "subscription"
      )
    },
    wsLink,
    httpLink
  ) : httpLink

  const cache = new InMemoryCache()

  const apolloClient = new ApolloClient({
    link: link,
    cache,
  })

  nuxtApp.vueApp.provide(provideApolloClient(apolloClient))
})
```

**Comment by Diizzayy** [comment, marked as answer]
@winnysan We've just released an early version of `@nuxtjs/apollo` v5, which enables all the features you'd expect ( including subscriptions of course ). It can be installed via `yarn add -D @nuxtjs/apollo@next`, you can have a look at the [documentation here](https://apollo.nuxtjs.org) ( It's actively being revised so minor bits may be subject to change )

## Demo Below

[![Open in StackBlitz](https://developer.stackblitz.com/img/open_in_stackblitz.svg)](https://stackblitz.com/edit/nuxt-apollo-demo)

**Reply by winnysan** [reply]
Excellent, that's phenomenal timing, thanks for the info @Diizzayy

**Reply by calebsmithdev** [reply]
Thank you for this! Much more useful than the hodged-podged version I created.
